fractions/javaee/ejb/src/main/java/org/wildfly/swarm/ejb/EJBFraction.java

```diff
@@ -56,32 +53,54 @@ public EJBFraction applyDefaults() {
         threadPoolSettings.put("time", "100");
         threadPoolSettings.put("unit", "MILLISECONDS");
 
-        defaultStatefulBeanAccessTimeout(5000L)
-                .defaultSingletonBeanAccessTimeout(5000L)
+        this
+                .defaultSlsbInstancePool("slsb-strict-max-pool")
+                .defaultStatefulBeanAccessTimeout(5000L)
                 .defaultSfsbCache("simple")
+                .defaultSfsbPassivationDisabledCache("simple")
+                .defaultSingletonBeanAccessTimeout(5000L)
+                .defaultResourceAdapterName(SwarmProperties.propertyVar("ejb.resource-adapter-name", "activemq-ra.rar"))
+                .defaultMdbInstancePool("mdb-strict-max-pool")
+                .strictMaxBeanInstancePool("slsb-strict-max-pool", pool -> {
+                    pool.deriveSize(StrictMaxBeanInstancePool.DeriveSize.FROM_WORKER_POOLS);
+                    pool.timeout(5L);
+                    pool.timeoutUnit(StrictMaxBeanInstancePool.TimeoutUnit.MINUTES);
+                })
+                .strictMaxBeanInstancePool("mdb-strict-max-pool", pool -> {
+                        pool.deriveSize(StrictMaxBeanInstancePool.DeriveSize.FROM_CPU_COUNT);
+                        pool.timeout(5L);
+                        pool.timeoutUnit(StrictMaxBeanInstancePool.TimeoutUnit.MINUTES);
+                })
+                .cache("simple")
+                // ideally, the distributable cache and the infinispan passivation store would be defined
+                // in InfinispanCustomizer, but that only applies after YAML is processed
+                .cache("distributable", cache -> {
+                    cache.alias("passivating");
+                    cache.alias("clustered");
+                    cache.passivationStore("infinispan");
+                })
+                .passivationStore("infinispan", store -> {
+                    store.cacheContainer("ejb");
+                    store.maxSize(10000);
+                })
```

Comment:
In a test, I have a `project-defaults.yml` file looking like this:

```yaml
swarm:
  ejb3:
    default-sfsb-cache: passivating
    passivation-stores:
      infinispan:
        max-size: 5
```

Since `project-defaults.yml` is processed _before_ the `Customizer`s, if I created the passivation store in the `InfinispanCustomizer` (where it naturally belongs), the code would actually try to create a 2nd passivation store with the same name. That obviously leads to deployment failure.

If there's something like `Customizer`s, but happening _before_ user config, that would be a perfect place for this.

Reply:
Can't we interrogate the current state of Infinispan to check for the passivation store?

Reply:
Would have to check what has been configured by the user, so that we don't overwrite user configuration, yet add everything that wasn't configured.